### DISCUSSION TITLE
How to update a particular field based on a variable

### DISCUSSION THREAD

**Discussion by jspann93** [pocketbase/pocketbase#2665] [Q&A] [answered]
I have a collection that stores records for a soccer lineup builder. For each player name, I want to keep track of how many times the player was selected overall and how those selections break down by position. For example, PlayerA was selected 5 times: 3 times as a striker, once as a winger, once as a midfielder. 

My thought it to do this by having a 'count' column that keeps track of the total number of selections and columns for each position (like in the image below). If I do that, how can I update the value of a particular column based on a variable?
<img width="1048" alt="image" src="https://github.com/pocketbase/pocketbase/assets/111022851/227e60eb-fe56-4aeb-8e69-550a87826367">

So in this example, 'count' is the name of a numeric column I want to increment but 'position' is a variable that holds a value such as "goalkeeper1", "leftback1", etc. If position == "goalkeeper1" I want to increment the value stored in the "goalkeeper1" column for that record. Is there a way to do this without a switch case and hard coding?
<img width="666" alt="image" src="https://github.com/pocketbase/pocketbase/assets/111022851/951726ec-892c-43bf-a8cf-ae2e233fd628">

**Comment by ganigeorgiev** [comment, marked as answer]
> If I do that, how can I update the value of a particular column based on a variable?

You can use the [bracket notation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_accessors#bracket_notation):

```js
const data = {
    count: existingRecord433.id,
};
data[position] = positionCount;

await pb.collection("men433").update(existingRecord433.id, data);
```

**Reply by jspann93** [reply]
That seems to be working. Thanks so much!!
